javascript/features/death_match/death_match_manager.js

```diff
@@ -29,38 +29,20 @@ export class DeathMatchManger {
 
         this.callbacks_ = new ScopedCallbacks();
         this.callbacks_.addEventListener(
-            'playerresolveddeath', DeathMatchManger.prototype.onPlayerDeath.bind(this));
-        this.callbacks_.addEventListener(
-            'playerspawn', DeathMatchManger.prototype.onPlayerSpawn.bind(this));
-        this.callbacks_.addEventListener(
-            'playerdisconnect', DeathMatchManger.prototype.onPlayerDisconnect.bind(this));
+            'playerdeath', DeathMatchManger.prototype.onPlayerDeath.bind(this));
```

Comment:
I think you really want to keep using `playerresolveddeath` here. Why did you change it back?

The difference is that we have a bunch of abuse and hammering checks that happen before `playerresolveddeath` is called. It's also faster, because we don't block the server on it.

Reply:
Actually, I see that the `MockPlayer.prototype.die` function doesn't account for that yet. That's a bug, which I fixed:

https://github.com/LVPlayground/playground/commit/c574e78686f084a2b62a24dde53bc03836a1e168